resources/aws/bucket.go

```diff
@@ -8,13 +8,18 @@ import (
 	"github.com/aws/aws-sdk-go/service/s3"
 	microerror "github.com/giantswarm/microkit/error"
 	"github.com/juju/errgo"
+
+	"github.com/giantswarm/aws-operator/resources"
 )
 
 type Bucket struct {
 	Name string
 	AWSEntity
 }
 
+// Implement ReusableResource.
+var _ = resources.ReusableResource(&Bucket{})
```

Comment:
Do we even need to confirm we conform to the interface in this case? Honest question.

If we're not using the interface type as an abstraction over the implementations anywhere (e.g. as an argument to a function) then what do we need the interface for?

If we call a method on a struct that's not there, we'll get a compile error anyway.

Or am I missing something obvious?